File: backend/penalty.py
from flask import Blueprint, request, jsonify

penalty_bp = Blueprint("penalty", __name__)

PENALTIES = {
    "cooking_in_hostel": {
        "label": "Cooking food in hostel room / using prohibited appliances",
        "penalties": {
            1: {
                "fine": 1000,
                "action": "Confiscation of equipment",
                "notes": "Parents informed"
            },
            2: {
                "fine": 5000,
                "action": "Non-returnable confiscation + written warning",
                "notes": "Parent undertaking required"
            },
            3: {
                "fine": 10000,
                "action": "Suspension from hostel",
                "notes": "May extend up to full academic tenure"
            }
        }
    },

    "alcohol": {
        "label": "Possession / consumption of alcohol on campus",
        "penalties": {
            1: {
                "fine": 5000,
                "action": "Written warning",
                "notes": "Parents informed"
            },
            2: {
                "fine": 10000,
                "action": "Proctorial warning + counselling",
                "notes": "Parent undertaking required"
            },
            3: {
                "fine": 20000,
                "action": "Suspension up to 1 year OR expulsion",
                "notes": "Scholarship / internship may be withheld"
            }
        }
    }
}


@penalty_bp.route("/calculate", methods=["POST"])
def calculate_penalty():
    data = request.json
    offence_key = data.get("offence")
    count = int(data.get("count", 1))

    if offence_key not in PENALTIES:
        return jsonify({"error": "Invalid offence"}), 400

    offence = PENALTIES[offence_key]
    level = min(count, 3)
    result = offence["penalties"][level]

    return jsonify({
        "offence": offence["label"],
        "occurrence": level,
        "fine": result["fine"],
        "action": result["action"],
        "notes": result["notes"],
        "source": "Bennett University Students’ Discipline & Conduct Rules",
        "disclaimer": "Final decision rests with the University Proctorial Committee."
    })

File: frontend/penalty/penalty.js
async function calculatePenalty() {
  const offence = document.getElementById("offence").value;
  const count = document.getElementById("count").value;
  const result = document.getElementById("result");

  if (!offence) {
    alert("Please select an offence.");
    return;
  }

  const res = await fetch("http://127.0.0.1:5000/penalty/calculate", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ offence, count })
  });

  const data = await res.json();

  if (data.error) {
    result.innerHTML = `<p>${data.error}</p>`;
    result.classList.remove("hidden");
    return;
  }

  result.innerHTML = `
    <h3>⚠️ Penalty Result</h3>
    <p><strong>Offence:</strong> ${data.offence}</p>
    <p><strong>Occurrence:</strong> ${data.occurrence}</p>
    <p><strong>Fine:</strong> ₹${data.fine}</p>
    <p><strong>Action:</strong> ${data.action}</p>
    <p><strong>Notes:</strong> ${data.notes}</p>
    <p style="font-size:12px;opacity:0.7">${data.source}</p>
  `;

  result.classList.remove("hidden");
}
